Crash when running /project command in assistant panel

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Launch Zed
2. Open the Assistant panel
3. Enter `/project` command and hit return
4. Application immediately crashes and closes

### Environment

Zed: v0.161.1 (Zed Preview)
OS: macOS 14.6.1
Memory: 18 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
[ERROR] {
  "thread": "main",
  "payload": "called `Result::unwrap()` on an `Err` value: no global registered of type semantic_index::SemanticDb",
  "location_data": {
    "file": "crates/assistant/src/slash_command/project_command.rs",
    "line": 95
  },
  "backtrace": [
    "<backtrace::capture::Backtrace>::create",
    "<backtrace::capture::Backtrace>::new",
    "zed::reliability::init_panic_hook::{closure#0}",
    "std::panicking::rust_panic_with_hook",
    "std::panicking::begin_panic_handler::{{closure}}",
    "std::sys::backtrace::__rust_end_short_backtrace",
    "_rust_begin_unwind",
    "core::panicking::panic_fmt",
    "core::result::unwrap_failed",
    "<gpui::app::AppContext>::lease_global::<semantic_index::SemanticDb>",
    "<assistant::slash_command::project_command::ProjectSlashCommand as assistant_slash_command::SlashCommand>::run",
    "<assistant::assistant_panel::ContextEditor>::run_command",
    "<assistant::slash_command::SlashCommandCompletionProvider>::complete_command_name::{closure#1}::{closure#0}::{closure#0}::{closure#0}::{closure#0}::{closure#0}",
    "<editor::Editor>::do_completion",
    "editor::element::register_action::<editor::actions::ConfirmCompletion, <editor::element::EditorElement>::register_actions::{closure#21}>::{closure#0}",
    "<gpui::window::WindowContext>::dispatch_action_on_node",
    "<gpui::window::WindowContext>::dispatch_event",
    "<gpui::window::Window>::new::{closure#10}",
    "gpui::platform::mac::window::handle_key_event",
    "<gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run",
    "<gpui::app::App>::run::<zed::main::{closure#5}>",
    "zed::main",
    "std::sys::backtrace::__rust_begin_short_backtrace::<fn(), ()>",
    "std::rt::lang_start::<()>::{closure#0}",
    "std::rt::lang_start_internal",
    "_main"
  ],
  "app_version": "0.162.0",
  "release_channel": "Zed Nightly",
  "os_name": "macOS",
  "os_version": "14.6.1",
  "architecture": "aarch64",
  "panicked_on": 1731461798232,
  "system_id": "e4f556c7-2751-48a8-b529-50779a285e14",
  "installation_id": "9a6f0846-72ca-4f1d-b9e3-251a56f5c267",
  "session_id": "37a9c0a5-fba0-46d6-81f4-016b465ecb5f"
}
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/assistant/src/assistant.rs
+++ b/crates/assistant/src/assistant.rs
@@ -18,6 +18,7 @@ mod terminal_inline_assistant;
 mod tool_working_set;
 mod tools;
 
+use crate::slash_command::project_command::ProjectSlashCommandFeatureFlag;
 pub use crate::slash_command_working_set::{SlashCommandId, SlashCommandWorkingSet};
 pub use crate::tool_working_set::{ToolId, ToolWorkingSet};
 pub use assistant_panel::{AssistantPanel, AssistantPanelEvent};
@@ -215,23 +216,32 @@ pub fn init(
         });
     }
 
-    if cx.has_flag::<SearchSlashCommandFeatureFlag>() {
-        cx.spawn(|mut cx| {
-            let client = client.clone();
-            async move {
-                let embedding_provider = CloudEmbeddingProvider::new(client.clone());
-                let semantic_index = SemanticDb::new(
-                    paths::embeddings_dir().join("semantic-index-db.0.mdb"),
-                    Arc::new(embedding_provider),
-                    &mut cx,
-                )
-                .await?;
+    cx.spawn(|mut cx| {
+        let client = client.clone();
+        async move {
+            let is_search_slash_command_enabled = cx
+                .update(|cx| cx.wait_for_flag::<SearchSlashCommandFeatureFlag>())?
+                .await;
+            let is_project_slash_command_enabled = cx
+                .update(|cx| cx.wait_for_flag::<ProjectSlashCommandFeatureFlag>())?
+                .await;
 
-                cx.update(|cx| cx.set_global(semantic_index))
+            if !is_search_slash_command_enabled && !is_project_slash_command_enabled {
+                return Ok(());
             }
-        })
-        .detach();
-    }
+
+            let embedding_provider = CloudEmbeddingProvider::new(client.clone());
+            let semantic_index = SemanticDb::new(
+                paths::embeddings_dir().join("semantic-index-db.0.mdb"),
+                Arc::new(embedding_provider),
+                &mut cx,
+            )
+            .await?;
+
+            cx.update(|cx| cx.set_global(semantic_index))
+        }
+    })
+    .detach();
 
     context_store::init(&client.clone().into());
     prompt_library::init(cx);

